Makefile

```diff
@@ -361,82 +279,133 @@ generate-go-conversions-kcp: $(CONVERSION_GEN)
 		--go-header-file=./hack/boilerplate/boilerplate.generatego.txt
 
+.PHONY: diagrams
+diagrams: ## Generate diagrams for *.plantuml files
+	$(MAKE) -C docs diagrams
 
-.PHONY: generate-manifests
-generate-manifests: $(addprefix generate-manifests-,$(ALL_GENERATE_MODULES)) ## Generate manifests e.g. CRD, RBAC etc.
+## --------------------------------------
+## Lint / Verify
+## --------------------------------------
 
-.PHONY: generate-manifests-core
-generate-manifests-core: $(CONTROLLER_GEN) $(KUSTOMIZE)
-	$(CONTROLLER_GEN) \
-		paths=./api/... \
-		paths=./controllers/... \
-		paths=./internal/webhooks/... \
-		paths=./$(EXP_DIR)/api/... \
-		paths=./$(EXP_DIR)/internal/controllers/... \
-		paths=./$(EXP_DIR)/addons/api/... \
-		paths=./$(EXP_DIR)/addons/internal/controllers/... \
-		crd:crdVersions=v1 \
-		rbac:roleName=manager-role \
-		output:crd:dir=./config/crd/bases \
-		output:webhook:dir=./config/webhook \
-		webhook
-	$(CONTROLLER_GEN) \
-		paths=./cmd/clusterctl/api/... \
-		crd:crdVersions=v1 \
-		output:crd:dir=./cmd/clusterctl/config/crd/bases
-	$(KUSTOMIZE) build $(CLUSTERCTL_MANIFEST_DIR)/crd > $(CLUSTERCTL_MANIFEST_DIR)/manifest/clusterctl-api.yaml
+##@ lint and verify:
 
-.PHONY: generate-manifests-cabpk
-generate-manifests-cabpk: $(CONTROLLER_GEN)
-	$(CONTROLLER_GEN) \
-		paths=./bootstrap/kubeadm/api/... \
-		paths=./bootstrap/kubeadm/internal/controllers/... \
-		crd:crdVersions=v1 \
-		rbac:roleName=manager-role \
-		output:crd:dir=./bootstrap/kubeadm/config/crd/bases \
-		output:rbac:dir=./bootstrap/kubeadm/config/rbac \
-		output:webhook:dir=./bootstrap/kubeadm/config/webhook \
-		webhook
+.PHONY: modules
+modules: ## Run go mod tidy to ensure modules are up to date
+	go mod tidy
+	cd $(TOOLS_DIR); go mod tidy
+	cd $(TEST_DIR); go mod tidy
 
-.PHONY: generate-manifests-kcp
-generate-manifests-kcp: $(CONTROLLER_GEN)
-	$(CONTROLLER_GEN) \
-		paths=./controlplane/kubeadm/api/... \
-		paths=./controlplane/kubeadm/controllers/... \
-		paths=./controlplane/kubeadm/internal/webhooks/... \
-		crd:crdVersions=v1 \
-		rbac:roleName=manager-role \
-		output:crd:dir=./controlplane/kubeadm/config/crd/bases \
-		output:rbac:dir=./controlplane/kubeadm/config/rbac \
-		output:webhook:dir=./controlplane/kubeadm/config/webhook \
-		webhook
+.PHONY: lint
+lint: $(GOLANGCI_LINT) ## Lint the codebase
+	$(GOLANGCI_LINT) run -v $(GOLANGCI_LINT_EXTRA_ARGS)
+	cd $(TEST_DIR); $(GOLANGCI_LINT) run -v $(GOLANGCI_LINT_EXTRA_ARGS)
+	cd $(TOOLS_DIR); $(GOLANGCI_LINT) run -v $(GOLANGCI_LINT_EXTRA_ARGS)
+
+.PHONY: lint-fix
+lint-fix: $(GOLANGCI_LINT) ## Lint the codebase and run auto-fixers if supported by the linter
+	GOLANGCI_LINT_EXTRA_ARGS=--fix $(MAKE) lint
+
+apidiff: $(GO_APIDIFF) ## Check for API differences
+	$(GO_APIDIFF) $(shell git rev-parse origin/main) --print-compatible
+
+.PHONY: format-tiltfile
+format-tiltfile: ## Format the Tiltfile
+	./hack/verify-starlark.sh fix
+
+.PHONY: verify
+verify: ## Run all verify-* targets
```

Comment:
What about adding `ALL_VERIFY_CHECKS` var and use `$(addprefix verify-,$(ALL_VERIFY_CHECKS))`

Reply:
+1 Added.